iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -186,10 +186,13 @@ inline bool vector<T, Capacity>::emplace(const uint64_t position, Targs&&... arg
     {
         return emplace_back(std::forward<Targs>(args)...);
     }
-
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
+    // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
     emplace_back(std::move(*reinterpret_cast<T*>(m_data[m_size - 1U])));
```

Comment:
I think in this method we could replace all `reinterpret_cast` with `at`. Then we also do not need to write axivion and clang-tidy suppressions.

Reply:
In this case I would suggest to introduce a private `at_unchecked`. There is no need to do the check at every access since we already ensured that the following operations work within the bounds of the vector.

Reply:
Fixed.